packages/x-data-grid-premium/src/DataGridPremium/useDataGridPremiumProps.ts

```diff
@@ -64,12 +65,11 @@ export const DATA_GRID_PREMIUM_PROPS_DEFAULT_VALUES: DataGridPremiumPropsWithDef
 const defaultSlots = DATA_GRID_PREMIUM_DEFAULT_SLOTS_COMPONENTS;
 
 export const useDataGridPremiumProps = (inProps: DataGridPremiumProps) => {
-  const themedProps =
-    // eslint-disable-next-line material-ui/mui-name-matches-component-name
-    useThemeProps({
```

Comment:
The returned value from `useThemeProps` is not memoized. See https://github.com/mui/material-ui/pull/45943